client/dashboard/sites/add-new-site/index.tsx

```diff
@@ -79,7 +79,7 @@ function AddNewSite() {
 						'Prompt, edit, and launch WordPress websites with Artificial Intelligence.'
 					) }
 					onClick={ bigSkyClick }
-					href="/setup/ai-site-builder"
+					href="/setup/ai-site-builder?source=add-new-site-popover"
```

Comment:
Not that this is a reusable component, it's not necessarily rendered in the "/sites" page, for instance in the new hosting dashboard, it will also be rendered in a site switcher dropdown. So updating the url here might not be sufficient in the long run depending on what we want the query string to do.